servers/rendering/renderer_rd/renderer_canvas_render_rd.h

```diff
@@ -46,6 +46,7 @@ class RendererCanvasRenderRD : public RendererCanvasRender {
 		MATERIAL_UNIFORM_SET = 1,
 		TRANSFORMS_UNIFORM_SET = 2,
 		CANVAS_TEXTURE_UNIFORM_SET = 3,
+		INSTANCE_DATA_UNIFORM_SET = 4,
```

Comment:
location 4 is already occupied by `uv_attrib`, I'm getting custom shader compilation errors with this PR.

Reply:
@alula `INSTANCE_DATA_UNIFORM_SET ` is a UBO for a descriptor set 4. `uv_attrib` is a vertex attribute at location 4. There is a vertex stage output variable at location 4:


```
layout(location = 4) out flat uint instance_index_interp;
```

Can you share the specific compilation errors

Reply:
Oh well I just did some debugging while making a MRP instead of doing it on large project and I realized I were confidently wrong :)

Well, turns out adding any `varying` to the shader triggers following compilation error:

```
ERROR: Failed parse:
WARNING: 0:270: '' : all default precisions are highp; use precision statements to quiet warning, e.g.:
         "precision mediump int; precision highp float;" 
ERROR: 0:270: 'location' : overlapping use of location 4
ERROR: 1 compilation errors.  No code generated.



   at: _compile_variant (servers/rendering/renderer_rd/shader_rd.cpp:294)
```

Quick look, let's see where are `varying` indices are being assigned by shader compiler...

https://github.com/godotengine/godot/blob/e96ad5af98547df71b50c4c4695ac348638113e0/servers/rendering/renderer_rd/renderer_canvas_render_rd.cpp#L2633

After changing that to `5` everything seems to work as expected.

Reply:
```glsl
shader_type canvas_item;

varying flat vec4 test1;

void vertex() {
	// Called for every vertex the material is visible on.
}

void fragment() {
	// Called for every pixel the material is visible on.
}
```

Reply:
@alula great find, thanks! Fixed in latest push